### PR TITLE
CodeClimate: update rules with newer version

### PR DESCRIPTION
**Pull request by jeherve**
#### Changes proposed in this Pull Request:

* We haven't updated our CodeClimate configuration in a while. It's now time to refresh things a bit, mostly by making sure we do not use any deprecated config flags, and that we use the currently supported version of the configuration tool.

#### Jetpack product discussion

* Related: p9dueE-1R7-p2

#### Does this pull request change what data or activity we track or use?

* No

#### Testing instructions:

* Check the CodeClimate tests below this PR; it shouldn't return any errors.

#### Proposed changelog entry for your changes:

* N/A

### CODE REVIEW

File: .github/codeclimate-base.yml [hunk 1]
```diff
@@ -9,7 +9,8 @@
 #   - url: "https://raw.githubusercontent.com/automattic/jetpack/master/.github/codeclimate-base.yml"
 #     path: "codeclimate-base.yml"
 ---
-engines:
+version: "2"
```
Comment (jeherve) [inline .github/codeclimate-base.yml:12]: See https://docs.codeclimate.com/docs/advanced-configuration#analysis-configuration-versions

File: .github/codeclimate-base.yml [hunk 2]
```diff
@@ -9,7 +9,8 @@
...
+plugins:
```
Comment (jeherve) [inline .github/codeclimate-base.yml:13]: See https://docs.codeclimate.com/docs/advanced-configuration#analysis-configuration-versions

> version: "2" is required at the top of the .codeclimate.yml.

> engines has been changed to plugins.

File: .github/codeclimate-base.yml [hunk 3]
```diff
@@ -28,23 +29,17 @@ engines:
     enabled: true
   eslint:
     enabled: true
-    channel: "eslint-5"
+    channel: "eslint-7"
     config:
       config: "/.eslintrc.js"
+      extensions:
+      - .js
+      - .jsx
```
Comment (jeherve) [inline .github/codeclimate-base.yml:37]: See https://docs.codeclimate.com/docs/eslint

File: .github/codeclimate-base.yml [hunk 4]
```diff
@@ -28,23 +29,17 @@ engines:
...
-  paths:
-  - "**.css"
-  - "**.scss"
-  - "**.inc"
-  - "**.js"
-  - "**.jsx"
-  - "**.php"
-  - "**.md"
-exclude_paths:
+exclude_patterns:
```
Comment (jeherve) [inline .github/codeclimate-base.yml:42]: > exclude_paths has been changed to exclude_patterns.
-- https://docs.codeclimate.com/docs/advanced-configuration#analysis-configuration-versions

File: .github/codeclimate-base.yml [hunk 5]
```diff
@@ -28,23 +29,17 @@ engines:
...
     enabled: true
-ratings:
-  paths:
-  - "**.css"
-  - "**.scss"
-  - "**.inc"
-  - "**.js"
-  - "**.jsx"
-  - "**.php"
-  - "**.md"
```
Comment (jeherve) [inline .github/codeclimate-base.yml:46]: > ratings has been deprecated. All files in our supported languages for maintainability will receive maintainability ratings by default.
-- https://docs.codeclimate.com/docs/advanced-configuration#analysis-configuration-versions

**Review by kraftbj — APPROVED** [review]

**Pull request merged**